azure/bastion/locals.tf

```diff
@@ -0,0 +1,3 @@
+locals {
+  whitelist = try(length(var.whitelist) == 0, false) ? ["Internet"] : var.whitelist
```

Comment:
`length(var.whitelist) >= 1 : var.whitelist : ["Internet"]` would be cleaner?

Or since it already has a default set in the variable, is this local even needed?

Reply:
Whitelist is set to `["Internet"]` by default but on the off chance you call the module with an empty list, this prevents a failure while being explicit about what the default is through the variable. Mostly done it this way because when I was calling it as a submodule of a submodule, the value being overwritten was failing. Happy to look into another way of going about it though.

Reply:
If the end user does pass in an empty list and knowingly overwrites default value, then should it be the modules responsibility to correct the user for their behaviour without the correction being highlighted? Could we not just use validation on the variable instead?

In the case of the module within the module, the parent module should have the check to ensure that the empty list cannot be passed in, I do not believe that this should be the responsibility of the child module to make the correction on behalf of the user as it may lead to frustration where the module is doing unexpected behaviour.